snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/EffectiveTimeRestorer.java

```diff
@@ -0,0 +1,503 @@
+
+				if (previousMapRule != null && !previousMapRule.equals(memberToRestore.getMapRule())) {
+					return false;
+				}
+
+				if (previousMapAdvice != null && !previousMapAdvice.equals(memberToRestore.getMapAdvice())) {
+					return false;
+				}
+
+				if (previousCorrelationId != null && !previousCorrelationId.equals(memberToRestore.getCorrelationId())) {
+					return false;
+				}
+				
+				// Handle extended map
+				if (memberToRestore.getMapCategoryId() != null && !memberToRestore.getMapCategoryId().equals(previousMapCategoryId)) {
+					return false;
+				}
+				
+				return memberToRestore.isActive() == previousMember.isActive() && memberToRestore.getModuleId().equals(previousMember.getModuleId());
+			} else if (componentToRestore instanceof SnomedSimpleMapRefSetMember) {
+				final SnomedSimpleMapRefSetMember memberToRestore = (SnomedSimpleMapRefSetMember) componentToRestore;
+				final String previousMapTargetDescription = (String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_TARGET_DESCRIPTION);
+				
+				if (memberToRestore.getMapTargetComponentDescription() != null && !memberToRestore.getMapTargetComponentDescription().equals(previousMapTargetDescription)) {
+					return false;
+				}
+				
+				return memberToRestore.isActive() == previousMember.isActive()
+						&& memberToRestore.getModuleId().equals(previousMember.getModuleId()) 
+						&& memberToRestore.getMapTargetComponentId().equals(previousMember.getProperties().get(SnomedRf2Headers.FIELD_MAP_TARGET));
+						
+			}
+			if (componentToRestore.eClass().equals(SnomedRefSetPackage.Literals.SNOMED_REF_SET_MEMBER)) {
+				// Simple type reference set
+				final SnomedRefSetMember memberToRestore = (SnomedRefSetMember) componentToRestore;
+				return memberToRestore.isActive() == previousMember.isActive() && memberToRestore.getModuleId().equals(previousMember.getModuleId());
+			} else {
+				return false;
+			}
+		} else {
+			throw new UnexpectedTypeException("Unexpected member type '" + componentToRestore.getClass() + "'.");
+		}
+	}
+	
+	private void restoreEffectiveTime(EObject componentToRestore, IComponent previousVersion) {
+		if (componentToRestore instanceof Component && previousVersion instanceof SnomedComponent) {
+			((Component) componentToRestore).setEffectiveTime(((SnomedComponent) previousVersion).getEffectiveTime());
+		} else if (componentToRestore instanceof SnomedRefSetMember && previousVersion instanceof SnomedComponent) {
+			((SnomedRefSetMember) componentToRestore).setEffectiveTime(((SnomedComponent) previousVersion).getEffectiveTime());
+		}
+	}
+	
+	private String getId(EObject object) {
+		if (object instanceof Component) {
+			final Component component = (Component) object;
+			return component.getId();
+		} else if (object instanceof SnomedRefSetMember) {
+			final SnomedRefSetMember member = (SnomedRefSetMember) object;
+			return member.getUuid();
+		}
+		
+		// XXX: Should never happen
+		throw new IllegalStateException();
```

Comment:
In this case, throw an `UnsupportedOperationException` or `IllegalArgumentException` with a proper message.